source/sigv4_quicksort.c

```diff
@@ -202,31 +201,32 @@ static size_t partition( void * pArray,
                          size_t itemSize,
                          ComparisonFunc_t comparator )
 {
-    void * pivot;
+    uint8_t * pivot;
+    uint8_t * pArrayLocal = ( uint8_t * ) pArray;
```

Comment:
This addresses the warnings generated (by `gcc-7.5.0`) for using arithmetic operations on `void *` 
```
/home/ubuntu/Repos/SigV4-for-AWS-IoT-embedded-sdk/source/sigv4_quicksort.c:212:20: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
     pivot = pArray + ( high * itemSize );
                    ^
/home/ubuntu/Repos/SigV4-for-AWS-IoT-embedded-sdk/source/sigv4_quicksort.c:220:32: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
         if( comparator( pArray + ( j * itemSize ), pivot ) < 0 )
                                ^
/home/ubuntu/Repos/SigV4-for-AWS-IoT-embedded-sdk/source/sigv4_quicksort.c:223:26: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
             swap( pArray + ( i * itemSize ), pArray + ( j * itemSize ), itemSize );
                          ^
/home/ubuntu/Repos/SigV4-for-AWS-IoT-embedded-sdk/source/sigv4_quicksort.c:223:53: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
             swap( pArray + ( i * itemSize ), pArray + ( j * itemSize ), itemSize );
                                                     ^
/home/ubuntu/Repos/SigV4-for-AWS-IoT-embedded-sdk/source/sigv4_quicksort.c:229:18: warning: pointer of type ‘void *’ used in arithmetic [-Wpointer-arith]
     swap( pArray + ( ( i + 1U ) * itemSize ), pivot, itemSize );
                  ^
```